src/mappings/staking.ts

```diff
@@ -97,28 +99,26 @@ export async function handleAddDelegation(
       amount: eraAmount,
       createdBlock: event.blockNumber,
     });
-
-    await updateTotalStake(
-      eraManager,
-      indexer,
-      amountBn,
-      'add',
-      event,
-      indexer === source
-    );
   } else {
     delegation.amount = await upsertEraValue(
       eraManager,
       delegation.amount,
       amountBn
     );
-
-    await updateTotalStake(eraManager, indexer, amountBn, 'add', event);
   }
 
+  await updateTotalStake(
+    eraManager,
+    indexer,
+    amountBn,
+    'add',
+    event,
+    indexer === source && !delegation
```

Comment:
Need `updateTotalStake` before check `delegation`, otherwise delegation will always be true